html_app/components/MicroscopyModelFactory.js

```diff
@@ -23,75 +23,73 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
-                                if(matches){
-                                    if(collection_id == '%CELL_LINE%'){
-                                        collection_id = cell_line;
-                                    }
-                                    if( scb.utils.isDefined(rule['use_collection_id'])){
-                                        collection_id = rule['use_collection_id'];
+
+                                alreadySelected = false;
+                                _.each(slide_array, function (x) {
+                                    if (_.contains(hash_list, x.hash)) {
+                                        alreadySelected = true;
                                     }
-                                    max = template.slide_parser[collection_id][slide_type][conditions].length;
-                                    number_of_comparisons = 0;
-                                    do{
-                                        /*Math.random returns num in [0,1) so no need to worry about subtracting 1 from index*/
-                                        index =  Math.floor(Math.random() * (max));
-                                        slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
-                                        alreadySelected = false;
-                                        _.each(slide_array, function(x){
-                                            if(_.contains(hash_list, x.hash)) {
-                                                alreadySelected = true;
-                                            }
-                                        });
-                                        number_of_comparisons += 1;
-                                    }while(alreadySelected && number_of_comparisons < max);
-                                    imgs=slide_array;
-                                    isFound = true;
-                                }else{
-                                    try {
-                                        max = template.slide_parser['default']['Dye']['HnE'].length;
-                                        index = Math.floor(Math.random() * (max));
-                                        imgs = template.slide_parser['default']['Dye']['HnE'][index];
-                                    } catch(e){}
-								}
-							}
-					});
-					state.slides = imgs;
-					state.slide_type = slide_type;
-				}
-			else if (parser.complex_parser){
+                                });
+                                number_of_comparisons += 1;
+                            } while (alreadySelected && number_of_comparisons < max);
+                            imgs = slide_array;
+                            isFound = true;
+                        }
+                    }
+                });
+                state.slides = imgs;
+                state.slide_type = slide_type;
+            }else if (parser.complex_parser){
```

Comment:
while we are in here, do you mind cleaning this be either just tabs, or just spaces, with a preference towards spaces